pallets/signet/src/lib.rs

```diff
@@ -0,0 +1,464 @@
+		SignatureRequested {
+			sender: T::AccountId,
+			payload: [u8; 32],
+			key_version: u32,
+			deposit: BalanceOf<T>,
+			chain_id: Vec<u8>,
+			path: Vec<u8>,
+			algo: Vec<u8>,
+			dest: Vec<u8>,
+			params: Vec<u8>,
+		},
+
+		/// Sign-respond request event
+		SignRespondRequested {
+			sender: T::AccountId,
+			transaction_data: Vec<u8>,
+			slip44_chain_id: u32,
+			key_version: u32,
+			deposit: BalanceOf<T>,
+			path: Vec<u8>,
+			algo: Vec<u8>,
+			dest: Vec<u8>,
+			params: Vec<u8>,
+			explorer_deserialization_format: u8,
+			explorer_deserialization_schema: Vec<u8>,
+			callback_serialization_format: u8,
+			callback_serialization_schema: Vec<u8>,
+		},
+
+		/// Signature response event
+		SignatureResponded {
+			request_id: [u8; 32],
+			responder: T::AccountId,
+			signature: Signature,
+		},
+
+		/// Signature error event
+		SignatureError {
+			request_id: [u8; 32],
+			responder: T::AccountId,
+			error: Vec<u8>,
+		},
+
+		/// Read response event
+		ReadResponded {
+			request_id: [u8; 32],
+			responder: T::AccountId,
+			serialized_output: Vec<u8>,
+			signature: Signature,
+		},
+	}
+
+	// ========================================
+	// Errors
+	// ========================================
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The pallet has already been initialized
+		AlreadyInitialized,
+		/// The pallet has not been initialized yet
+		NotInitialized,
+		/// Unauthorized - caller is not admin
+		Unauthorized,
+		/// Insufficient funds for withdrawal
+		InsufficientFunds,
+		/// Invalid transaction data (empty)
+		InvalidTransaction,
+		/// Arrays must have the same length
+		InvalidInputLength,
+		/// The chain ID is too long
+		ChainIdTooLong,
+	}
+
+	// ========================================
+	// Extrinsics
+	// ========================================
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Initialize the pallet with admin, deposit, and chain ID
+		#[pallet::call_index(0)]
+		#[pallet::weight(<T as Config>::WeightInfo::initialize())]
+		pub fn initialize(
+			origin: OriginFor<T>,
+			admin: T::AccountId,
+			signature_deposit: BalanceOf<T>,
+			chain_id: BoundedVec<u8, T::MaxChainIdLength>,
+		) -> DispatchResult {
+			let _initializer = ensure_signed(origin)?;
+			ensure!(Admin::<T>::get().is_none(), Error::<T>::AlreadyInitialized);
+
+			Admin::<T>::put(&admin);
+			SignatureDeposit::<T>::put(signature_deposit);
+
+			let bounded_chain_id = BoundedVec::<u8, T::MaxChainIdLength>::try_from(chain_id.clone())
+				.map_err(|_| Error::<T>::ChainIdTooLong)?;
+			ChainId::<T>::put(bounded_chain_id);
+
+			Self::deposit_event(Event::Initialized {
+				admin,
+				signature_deposit,
+				chain_id: chain_id.to_vec(),
+			});
+
+			Ok(())
+		}
+
+		/// Update the signature deposit amount (admin only)
+		#[pallet::call_index(1)]
+		#[pallet::weight(<T as Config>::WeightInfo::update_deposit())]
+		pub fn update_deposit(origin: OriginFor<T>, new_deposit: BalanceOf<T>) -> DispatchResult {
```

Comment:
as dicusssed in other channels - we need probably to set some max limit that this deposit can be set to.